### PR TITLE
Replace the Person class with Subject

### PR DESCRIPTION
**Pull request by hokwaichan**
# Ticket Link

[1654](https://uhawaii.atlassian.net/browse/GROUPINGS-1654)

# List of squashed commits

- Replace Person with Subject

# Test Checklist

- [x] Exhibits Clear Code Structure:
- [x] Project Unit Tests Passed:
- [x] Project Jasmine Tests Passed:
- [x] Executes Expected Functionality:
- [x] Tested For Incorrect/Unexpected Inputs:

### CODE REVIEW

**Review by mhodgesatuh — COMMENTED** [review]
There appear to be errors flagged by the PR request that first need to be addressed.

File: src/main/resources/messages.properties
Comment (JorWo) [inline src/main/resources/messages.properties:1]: We want to keep the naming as usernames and persons for the UI because that is the terminology that the general audience is used to. This only applies to the text that the user can see, but not the code underneath. Could you revert the changes for this file? Thanks.

File: src/main/resources/grouper.client.base.properties [hunk 1]
```diff
@@ -525,7 +525,7 @@ esb.consumer.encryptionImplementation = edu.internet2.middleware.grouperClient.e
 ## general xmpp configuration
 grouperClient.xmpp.server.host = jabber.school.edu
 grouperClient.xmpp.server.port = 5222
-grouperClient.xmpp.user = username
+grouperClient.xmpp.user = uid
```
Comment (JorWo) [inline src/main/resources/grouper.client.base.properties:528]: I don't know if these properties are being used here, but keep it as username because these properties are completely different from the scope of the ticket.

**Comment by JorWo** [comment]
Could you revert all the text and aria attributes where in HTML files where username was replaced with uid? Thanks. The average user understands the term username but not uid. The uid and uhUuid convention is for the code only.

File: src/main/resources/messages.properties [hunk 2]
```diff
@@ -142,7 +142,7 @@ screen.message.common.tooltip.info=Click for more Information
 screen.message.common.tooltip.export.csv=Export all members to a list. Use the menu on the left to select the set of members to be exported.
 screen.message.common.tooltip.export.csv.large=Export will be available once all members are loaded
 screen.message.common.membership.info=Membership changes made may not take effect immediately. Usually, 3-5 minutes should be anticipated. In extreme cases changes may take several hours to be fully processed, depending on the number of members and the synchronization destination.
-screen.message.common.tooltip.uid.notApplicable=UH Username not available. Either it has not yet been assigned, or the person is no longer with UH.
+screen.message.common.tooltip.username.notApplicable=UH Username not available. Either it has not yet been assigned, or the subject is no longer with UH.
```
Comment (JorWo) [inline src/main/resources/messages.properties:145]: Keep this as person, thanks.

File: src/main/java/edu/hawaii/its/groupings/access/UhAttributes.java [hunk 3]
```diff
@@ -94,7 +65,7 @@ public List<String> getMail() {
     }
 
     public List<String> getAffiliation() {
-        return getValues("eduPersonAffiliation");
+        return getValues("eduSubjectAffiliation");
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/groupings/access/UhAttributes.java:68]: The `getAffliation` should be the original `eduPersonAffiliation`. Make sure to update the test too, thanks.

File: src/main/java/edu/hawaii/its/groupings/access/User.java [hunk 4]
```diff
@@ -20,9 +20,7 @@ public User(String uid, Collection<GrantedAuthority> authorities) {
         super(uid, "", authorities);
     }
 
-    public String getUid() {
-        return getUsername();
-    }
+    public String getUid() { return getUsername(); }
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/groupings/access/User.java:23]: Please expand out the curly braces so that:
```
public String getUid() {
    return getUsername();
}
```

File: src/main/java/edu/hawaii/its/groupings/access/User.java [hunk 5]
```diff
@@ -38,8 +36,7 @@ public UhAttributes getAttributes() {
         return attributes;
     }
 
-    public void setAttributes(UhAttributes attributes) { this.attributes = attributes;
-    }
+    public void setAttributes(UhAttributes attributes) { this.attributes = attributes; }
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/groupings/access/User.java:39]: Also expand the curly braces here, thanks!

File: src/main/resources/static/javascript/mainApp/general.controller.js [hunk 6]
```diff
@@ -14,7 +14,7 @@
      */
 
     function GeneralJsController($scope, $controller, $window, $uibModal, groupingsService, dataProvider, Message) {
-        // This is a regex Pattern that contains all valid UH Identifiers which consists of uid (Username) and uhUuid (UH Numbers) chars.
+        // This is a regex Pattern that contains all valid UH Identifiers which consists of uid (Uid) and uhUuid (UH Numbers) chars.
```
Comment (JorWo) [inline src/main/resources/static/javascript/mainApp/general.controller.js:17]: The `(Uid)` in the comment should stay as `(Username)` because the comment explains what is a uid and uhUuid.

File: src/main/resources/static/javascript/mainApp/grouping.controller.js [hunk 7]
```diff
@@ -1765,7 +1765,7 @@
          * @returns string table in CSV format
          */
         $scope.convertListToCsv = (table) => {
-            let str = "Last,First,Username,UH Number,Email\r\n";
+            let str = "Last,First,Uid,UH Number,Email\r\n";
```
Comment (JorWo) [inline src/main/resources/static/javascript/mainApp/grouping.controller.js:1768]: This should stay as `Username` because the average user is used to username, thanks.

File: src/main/resources/templates/admin.html [hunk 8]
```diff
@@ -20,7 +20,7 @@ <h1 class="text-center text-md-left">UH Groupings Administration</h1>
                         Manage Admins
                     </a>
                 <li class="nav-item" role="presentation">
-                    <a href="#manage-person" id="manage-person-tab" data-toggle="tab" class="nav-link tab-color" role="tab">
+                    <a href="#manage-subject" id="manage-subject-tab" data-toggle="tab" class="nav-link tab-color" role="tab">
                         Manage Person
```
Comment (JorWo) [inline src/main/resources/templates/admin.html:24]: Since the Manage Person tab is only used by a small number of people, let's go ahead and rename the text as well to Manage Subject.

File: src/test/javascript/grouping.controller.test.js [hunk 9]
```diff
@@ -2767,7 +2767,7 @@ describe("GroupingController", () => {
 
             scope.exportGroupToCsv(scope.groupingMembers, "members");
 
-            expect(mockElement.href).toBe("data:text/csv;charset=utf-8,Last,First,Username,UH%20Number,Email%0D%0AOne,User,user1,00000001,user1@hawaii.edu%0D%0ATwo,User,user2,00000002,user2@hawaii.edu%0D%0AThree,User,user3,00000003,user3@hawaii.edu%0D%0ASeven,User,user7,00000007,user7@hawaii.edu%0D%0A");
+            expect(mockElement.href).toBe("data:text/csv;charset=utf-8,Last,First,Uid,UH%20Number,Email%0D%0AOne,User,user1,00000001,user1@hawaii.edu%0D%0ATwo,User,user2,00000002,user2@hawaii.edu%0D%0AThree,User,user3,00000003,user3@hawaii.edu%0D%0ASeven,User,user7,00000007,user7@hawaii.edu%0D%0A");
```
Comment (JorWo) [inline src/test/javascript/grouping.controller.test.js:2770]: The `Uid` will have to be changed back to `Username` here too.

File: src/test/javascript/grouping.controller.test.js [hunk 10]
```diff
@@ -2777,7 +2777,7 @@ describe("GroupingController", () => {
             it("should start with the correct column headers", () => {
                 const csv = scope.convertListToCsv(scope.groupingExclude);
 
-                expect(csv.indexOf("Last,First,Username,UH Number,Email\r\n")).toEqual(0);
+                expect(csv.indexOf("Last,First,Uid,UH Number,Email\r\n")).toEqual(0);
```
Comment (JorWo) [inline src/test/javascript/grouping.controller.test.js:2780]: The `Uid` will also have to be changed back to `Username` here too.

File: uh-groupings-ui-overrides.skeleton.properties [hunk 11]
```diff
@@ -1,4 +1,4 @@
-# Create a directory in your home directory called ".<username for your computer>-conf"
+# Create a directory in your home directory called ".<uid for your computer>-conf"
```
Comment (JorWo) [inline uh-groupings-ui-overrides.skeleton.properties:1]: Keep this as `username` since the comment is talking about your computer's username, not a subject.

**Review by JorWo — APPROVED** [review]

**Review by mhodgesatuh — APPROVED** [review]
